runtime/compiler/control/J9CompilationStrategy.cpp

```diff
@@ -39,95 +39,26 @@
 #include "env/ut_j9jit.h"
 #include "env/CompilerEnv.hpp"
 
-// NOTE: TR::CompilationController is actually defined in control/OptimizationPlan.hpp
-
-TR::CompilationStrategy *TR::CompilationController::_compilationStrategy = NULL;
-TR::CompilationInfo *    TR::CompilationController::_compInfo = 0;
-int32_t                  TR::CompilationController::_verbose = 0;
-bool                     TR::CompilationController::_useController = false;
-bool                     TR::CompilationController::_tlsCompObjCreated = false;
-
-
-//------------------------------------ init -----------------------------------
-// Initializes the compilationController.
-// Return false if it fails
-//-----------------------------------------------------------------------------
-bool TR::CompilationController::init(TR::CompilationInfo *compInfo)
-   {
-   _useController = false; // Default to failure
-   _compilationStrategy = 0; // Default to failure
-   TR::Options *options = TR::Options::getCmdLineOptions();
-   char *strategyName = options->getCompilationStrategyName();
-
-   if (strategyName && strcmp(strategyName, "none"))
-      {
-      _compInfo = compInfo;
-      if (strcmp(strategyName, "default") == 0)
-         _compilationStrategy = new (PERSISTENT_NEW) TR::DefaultCompilationStrategy();
-      else if (strcmp(strategyName, "threshold") == 0)
-         _compilationStrategy = new (PERSISTENT_NEW) TR::ThresholdCompilationStrategy();
-      else // if no match, use default
-      {
-         _compilationStrategy = new (PERSISTENT_NEW) TR::DefaultCompilationStrategy();
-      }
-
-      if (_compilationStrategy)
-         {
-         TR_OptimizationPlan::_optimizationPlanMonitor = TR::Monitor::create("OptimizationPlanMonitor");
-         _useController = (TR_OptimizationPlan::_optimizationPlanMonitor != 0);
-         if (_useController)
-            {
-            static char *verboseController = feGetEnv("TR_VerboseController");
-            if (verboseController)
-               setVerbose(atoi(verboseController));
-            if (verbose() >= LEVEL1)
-               fprintf(stderr, "Using %s comp strategy\n", strategyName);
-            }
-         }
-      }
-   //TR_ASSERT(_useController, "Must use compilation controller");
-//#ifdef COMP_YIELD_ANALYSIS
-   if (options->getOption(TR_EnableCompYieldStats))
-      TR::Compilation::allocateCompYieldStatsMatrix();
-   tlsAlloc(OMR::compilation);
-   _tlsCompObjCreated = true;
-   return _useController;
-   }
-
-
-//-------------------------------- shutdown ---------------------------------
-// Called at shutdown time after compilation thread has been stopped
-// --------------------------------------------------------------------------
-void TR::CompilationController::shutdown()
+TR::CompilationStrategy*
+J9::CompilationStrategy::createInstance(const char* name)
```

Comment:
I would need to look at the OMR changes to see if this enough. Please add a link to the omr PR.

Reply:
dependent OMR change is https://github.com/eclipse/omr/pull/6884